pkg/api/kptfile/v1alpha2/types.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package defines the schema for Kptfile version v1alpha2.
+package v1alpha2
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	KptFileName       = "Kptfile"
+	KptFileGroup      = "kpt.dev"
+	KptFileVersion    = "v1alpha2"
+	KptFileAPIVersion = KptFileGroup + "/" + KptFileVersion
+)
+
+// KptFile contains information about a package managed with kpt.
+type KptFile struct {
+	yaml.ResourceMeta `yaml:",inline"`
+
+	// UpstreamLock is a resolved locator for the last fetch of the package.
+	UpstreamLock *UpstreamLock `yaml:"upstreamLock,omitempty"`
```

Comment:
seems like we had Upstream and now this is UpstreamLock.  Does it have the meaning of a lock?  Lock in NPM says that unless I update the refresh grabs the same versions.  Is that what we are doing and is that why we decided on a name change?

Reply:
There's a discussion below, but effectively yes. `UpstreamLock` is product of kpt resolving the user-specified `Upstream`.